### PR TITLE
bashorg robot

### PR DESCRIPTION
**Pull request by kirylpl**

### CODE REVIEW

File: bot.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require 'nokogiri'
+require 'telegram/bot'
+require 'open-uri'
+
+token = '369433381:AAF4mnXcr1Zp_KQ_5So5BZ2B1pLxmD5yhCw'
```
Comment (anatoliliotych) [inline bot.rb]: move to .env file and use dotenv gem

File: bot.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+$posts = []
+
+# Parse method
+def parse_posts
+  page = Nokogiri::HTML(open('http://bash.im/best'))
```
Comment (anatoliliotych) [inline bot.rb]: move url to constant

File: bot.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+parse_posts
+
+# Bot runner
+Telegram::Bot::Client.run(token) do |bot|
+  begin
+    bot.listen do |message|
+      case message.text
+      when '/posts'
+        bot.api.send_message(
```
Comment (anatoliliotych) [inline bot.rb]: check style guide! idents are wrong

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+TG_TOKEN=369433381:AAF4mnXcr1Zp_KQ_5So5BZ2B1pLxmD5yhCw
```
Comment (anatoliliotych) [inline .env]: and .env to .gitignore
and .env.example for everyone who wants to use your bot.

File: bot.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+require 'nokogiri'
+require 'telegram/bot'
+require 'open-uri'
+
+token = ENV['TG_TOKEN']
+$posts = []
+
+# Parse method
+def parse_posts
+  url = 'http://bash.im/best'
```
Comment (anatoliliotych) [inline bot.rb]: it's not a constant

File: bot.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        /refresh - get freshest posts \n
+        /stop - stop recieving joke posts") 
+      else
+        bot.api.send_message(
+        chat_id: message.chat.id,
+        text: "Sorry, IDK what you want from me")
+      end
+    end
+  rescue Telegram::Bot::Exceptions::ResponseError => e
+    retry
```
Comment (anatoliliotych) [inline bot.rb]: it's better to log errors

File: bot.rb (hunk 3 above)
Comment (kirylpl) [inline bot.rb]: @anatoliliotych rubocop now says idents are okay

File: bot.rb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          chat_id: message.chat.id,
+          text:  "Hi, #{message.from.first_name}.\n
+          Total posts: #{$posts.length}"
+        )
+        bot.api.send_message(
+          chat_id: message.chat.id,
+          # don't get how to insert '/n' to this yet
+          text: $posts.first(10).join("\n \n").to_s
+        )
+        # machinegun
```
Comment (anatoliliotych) [inline bot.rb]: don't leave such things in code. if this code doesn't work and you don't need it - use gist.github.com :)

File: bot.rb [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        )
+      else
+        bot.api.send_message(
+          chat_id: message.chat.id,
+          text: 'Sorry, IDK what you want from me'
+        )
+      end
+    end
+  rescue Telegram::Bot::Exceptions::ResponseError => err
+    logger = Logger.new('logfile.log')
```
Comment (anatoliliotych) [inline bot.rb]: create logger and pass it to bot during initialization

**Comment by anatoliliotych** [comment]
Not sure why it's merged - you didn't reposnd to code review...

**Comment by kirylpl** [comment]
@anatoliliotych Kirill asked to deploy bots, so I pushed to master as is.

File: .env (hunk 4 above)
Comment (kirylpl) [inline .env]: @anatoliliotych usyo dobra now.

**Comment by anatoliliotych** [comment]
did he asked to merge to master?

**Comment by kirylpl** [comment]
@anatoliliotych No, just wanted to set up automatic deploy from master and never look it back by myself

**Comment by anatoliliotych** [comment]
So please open PR for review.

**Pull request merged**